### PR TITLE
Add emit and builder for TSImportType

### PR DESCRIPTION
**Pull request by hzoo**
Authors: @existentialism and I

| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | Fixes https://github.com/babel/babel/issues/7749, Closes https://github.com/babel/babel/issues/7747
| Patch: Bug Fix?          |
| Major: Breaking Change?  |
| Minor: New Feature?      | 👍
| Tests Added + Pass?      | Yes
| Documentation PR Link    | 
| Any Dependency Changes?  |
| License                  | MIT

### CODE REVIEW

File: packages/babel-types/src/definitions/typescript.js [hunk 1]
```diff
@@ -405,6 +405,16 @@ defineType("TSModuleBlock", {
   },
 });
 
+defineType("TSImportType", {
+  aliases: ["TSType"],
+  visitor: ["argument", "qualifier", "typeParameters"],
+  fields: {
+    argument: validateType("StringLiteral"),
+    qualifier: validateOptionalType("TSEntityName"),
+    typeParameters: validateOptionalType("TSTypeParameterDeclaration"),
```
Comment (armano2) [inline packages/babel-types/src/definitions/typescript.js]: `TSTypeParameterInstantiation`

**Comment by armano2** [comment]
in types you should also update `TSTypeQuery`

https://github.com/babel/babel/blob/060f02ace5e362ae9e00b379dfbc261892be0ef5/packages/babel-types/src/definitions/typescript.js#L188
```
    exprName: validateType("TSEntityName", "TSImportType"),
```

**Comment by nicolo-ribaudo** [comment]
(OT)

> Authors: @existentialism and I

GitHub supports marking a commitment as done by multiple autors 😉

**Comment by hzoo** [comment]
Haha good point @nicolo-ribaudo, we talked about it on the stream too

**Pull request merged**